processor.go

```diff
@@ -282,10 +345,15 @@ func (pro *Processor) loopVote(candidate *base.Vertex) error {
 		return nil
 	}
 
+	// up to this step - the current instance is a collector
+	// the node signs the candidate vertex with its key and sends out its vote for it
+	// using the Signer interface implementation
+	// after the vote, the node processes its own vote immediately
+
 	// finally, if we are not the proposer, but the collector, create our own
 	// vote for the proposal and immediately process it locally to give it
 	// priority and make sure a proposal is generated if our own vote is the
-	// only one required for a qualified majority
+	// only one required for a qualified majority - I do not understand this part
```

Comment:
If we are responsible for collecting votes for a previous proposal, we also need to vote on it, and forward our vote to the next leader/proposer. Otherwise, we might not include our own vote (not get rewards) if everyone voted before us, and also, if we are in a scenario where a single node can be the majority needed to approve a block (i.e. 1/3 with 3 nodes, or a 1-node network), we have to do this for the protocol to make progress.